CAP/examples/unwrapped_MatrixCategory.g

```diff
@@ -0,0 +1,184 @@
+#! @Chapter Examples and Tests
+
+#! @Section WrapperCategory
+
+#! A priori, CAP only supports *wrapped* categories, that is, objects and morphisms in a category cannot be represented by primitive data types (like integers) but must be attribute storing GAP objects.
+#! In theory, CAP could support *bare* categories, that is, categories where objects and morphisms are represented by primitive data types. However, this would not be very convenient:
+#! * Type checking would be severely limited due to missing type information. For example, one could not distinguish between objects in a category of rows and objects in a category of columns if both are represented by plain integers.
+#! * In particular, one would always have to call CAP operations with the category as the first argument.
+#! * Morphisms could not store their source and range, so all CAP operations would have to be extended to explicitly include source and range of the morphisms in the input in addition to the primitive morphism data.
+#! * In particular, users would have to explicitly compute and take care of keeping records of the correct source and range of morphisms. In some cases this might actually lead to optimizations, but in most cases users would probably develop their own data structure for storing source and range together with the morphism datum, thus inventing the data structure we wanted to avoid in the first place.
+#! Despite these impracticalities in algorithms, when thinking about mathematics we often deal with categories where the objects and morphisms are of some primitive data type and the category is clear from the context.
+#! In particular, in principle we actually have to proof that our wrapping and unwrapping in CAP defines an equivalence of categories. Often this is trivial because wrapping and unwrapping just cancels.
```

Comment:
```suggestion
#! In particular, in principle, we actually have to prove that our wrapping and unwrapping in CAP defines an equivalence of categories. Often this is trivial because wrapping and unwrapping just cancels.
```

Reply:
Thanks!